### PR TITLE
fix(client): tighten test evaluator

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
- fix: prevent __utils posting arbitrary messages
- fix: freeze test-evaluator objects

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

File: client/src/client/workers/test-evaluator.ts [hunk 1]
```diff
@@ -111,6 +104,10 @@ ctx.onmessage = async (e: TestEvaluatorEvent) => {
 
   const assert = chai.assert;
   const __helpers = helpers;
+  // If the learner tries to modify these, weird behavior may result. Freezing
+  // them means they get an error instead.
+  Object.freeze(assert);
+  Object.freeze(__helpers);
```
Comment (ShaunSHamilton) [inline client/src/client/workers/test-evaluator.ts:110]: Why do we not also freeze `__utils`?

Comment (ojeytonwilliams) [reply client/src/client/workers/test-evaluator.ts:110]: It's frozen, but at the end of the file. I'll move it up so it's more obvious.

Comment (ShaunSHamilton) [reply client/src/client/workers/test-evaluator.ts:110]: 🤦‍♂️

**Review by ShaunSHamilton — APPROVED** [review]
DWINTW 👍

**Review by naomi-lgbt — APPROVED** [review]

**Pull request merged**